src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -23,16 +26,18 @@
  */
 class SendersLocator implements SendersLocatorInterface
 {
-    private $senders;
+    private $sendersMap;
+    private $sendersLocator;
     private $sendAndHandle;
 
     /**
-     * @param SenderInterface[][] $senders
-     * @param bool[]              $sendAndHandle
+     * @param string[] $sendersMap
+     * @param bool[]   $sendAndHandle
      */
-    public function __construct(array $senders, array $sendAndHandle = [])
+    public function __construct(array $sendersMap, ContainerInterface $sendersLocator, array $sendAndHandle = [])
```

Comment:
Instead of each sender class having its own service locator object of senders, this `$sendersMap` is now a simple array of string sender aliases (or service ids, if no alias) and `$sendersLocator` is a service locator that contains all senders.